### PR TITLE
1544 add remove vc manually

### LINKED ISSUE #1544: Add methods to add/remove entries of VCRegistry manually
### Context

Our R1 is planned on staging-parachain, we need a way to take the snapshot (state) of the VCRegistry of the staging-parachain when R1 ends or whenever the parachain is completely reset, and manually populate the VCRegistry of the litentry-rococo parachain with the snapshot.

To do that we need to add extrinsics that do can that.
Ideally we can test the workflow once

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by wangminqi**
This closes #1544

### CODE REVIEW

File: pallets/vc-management/src/lib.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@ pub mod pallet {
 		type SetAdminOrigin: EnsureOrigin<Self::RuntimeOrigin>;
 	}
 
-	// a map VCIndex -> VC context
+	// a map VCIndex -> VC contextca
```
Comment (Kailai-Wang) [inline pallets/vc-management/src/lib.rs]: typo

File: pallets/vc-management/src/lib.rs [hunk 2]
```diff
@@ -368,5 +376,42 @@ pub mod pallet {
 			Self::deposit_event(Event::SchemaRevoked { account: sender, shard, index });
 			Ok(().into())
 		}
+
+		/// ---------------------------------------------------
+		/// The following extrinsics are supposed to be called by Sudo/Concuil only
```
Comment (Kailai-Wang) [inline pallets/vc-management/src/lib.rs]: typo: Council

File: pallets/vc-management/src/lib.rs [hunk 3]
```diff
@@ -368,5 +376,42 @@ pub mod pallet {
...
+		/// This is the temporary function for manual operation of VCRegistry storage
+		/// ---------------------------------------------------
+		#[pallet::call_index(10)]
+		#[pallet::weight(195_000_000)]
+		pub fn add_vc(
```
Comment (Kailai-Wang) [inline pallets/vc-management/src/lib.rs]: Maybe you missed this comment: https://github.com/litentry/litentry-parachain/issues/1544#issuecomment-1495589922

I think `add_vcregistry_item` is a better name, as we are not adding VCs here (we don't have VC content), but a VCRegistry item.

File: pallets/vc-management/src/lib.rs [hunk 4]
```diff
@@ -368,5 +376,42 @@ pub mod pallet {
...
+			assertion: Assertion,
+			hash: H256,
+		) -> DispatchResultWithPostInfo {
+			T::SetAdminOrigin::ensure_origin(origin)?;
+			ensure!(!VCRegistry::<T>::contains_key(index), Error::<T>::VCAlreadyExists);
+			VCRegistry::<T>::insert(
+				index,
+				VCContext::<T>::new(subject.clone(), assertion.clone(), hash),
+			);
+			Self::deposit_event(Event::VCRegistryManualAdded {
```
Comment (Kailai-Wang) [inline pallets/vc-management/src/lib.rs]: Maybe: `VCRegistryItemAdded`

File: pallets/vc-management/src/lib.rs [hunk 5]
```diff
@@ -368,5 +376,42 @@ pub mod pallet {
...
+
+		#[pallet::call_index(11)]
+		#[pallet::weight(195_000_000)]
+		pub fn remove_vc(origin: OriginFor<T>, index: VCIndex) -> DispatchResultWithPostInfo {
+			T::SetAdminOrigin::ensure_origin(origin)?;
+			let _ = VCRegistry::<T>::get(index).ok_or(Error::<T>::VCNotExist)?;
+			VCRegistry::<T>::remove(index);
+			Self::deposit_event(Event::VCRegistryManualRemoved { index });
+			Ok(().into())
+		}
```
Comment (Kailai-Wang) [inline pallets/vc-management/src/lib.rs:425]: let's have another extrinsic to remove_all -> `clear()` is enough

**Review by Kailai-Wang — COMMENTED** [review]
Thank you for the quick engagement!

Several very small things:

File: pallets/vc-management/src/lib.rs (hunk 3 above)
Comment (wangminqi) [reply pallets/vc-management/src/lib.rs]: changed.

File: pallets/vc-management/src/lib.rs (hunk 4 above)
Comment (wangminqi) [reply pallets/vc-management/src/lib.rs]: changed.

File: pallets/vc-management/src/lib.rs (hunk 5 above)
Comment (wangminqi) [reply pallets/vc-management/src/lib.rs:425]: added

**Review by Kailai-Wang — APPROVED** [review]
Thanks - there're still some typos 😂 

I'll fix them in the upcoming PR

**Pull request merged**